src/node/internal/internal_fs_callback.ts

```diff
@@ -0,0 +1,1373 @@
+import * as fssync from 'node-internal:internal_fs_sync';
+import type {
+  CopySyncOptions,
+  OpenDirOptions,
+  MkdirSyncOptions,
+  MkdirTempSyncOptions,
+  ReadDirOptions,
+  ReadDirResult,
+  ReadFileSyncOptions,
+  ReadLinkSyncOptions,
+  ReadSyncOptions,
+  RmdirSyncOptions,
+  RmSyncOptions,
+  StatOptions,
+  StatFs,
+  WriteFileOptions,
+  WriteSyncOptions,
+} from 'node-internal:internal_fs_sync';
+import {
+  validatePosition,
+  type FilePath,
+} from 'node-internal:internal_fs_utils';
+import { F_OK } from 'node-internal:internal_fs_constants';
+import {
+  ERR_INVALID_ARG_TYPE,
+  ERR_INVALID_ARG_VALUE,
+} from 'node-internal:internal_errors';
+import { Stats } from 'node-internal:internal_fs_utils';
+import { type Dir } from 'node-internal:internal_fs';
+import { Buffer } from 'node-internal:internal_buffer';
+import { isArrayBufferView } from 'node-internal:internal_types';
+import { validateUint32 } from 'node-internal:validators';
+
+type ErrorOnlyCallback = (err: unknown) => void;
+type SingleArgCallback<T> = (err: unknown, result?: T) => void;
+type DoubleArgCallback<T, U> = (err: unknown, result1?: T, result2?: U) => void;
+
+function callWithErrorOnlyCallback(
+  fn: () => void,
+  callback: undefined | ErrorOnlyCallback
+): void {
+  if (typeof callback !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('callback', ['function'], callback);
+  }
+  try {
+    fn();
+    // Note that any errors thrown by the callback will be "handled" by passing
+    // them along to the reportError function, which logs them and triggers the
+    // global "error" event.
+    // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
+    queueMicrotask(() => callback(null));
+  } catch (err) {
+    // eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
+    queueMicrotask(() => callback(err));
+  }
+}
+
+function callWithSingleArgCallback<T>(
+  fn: () => T,
+  callback: undefined | SingleArgCallback<T>
+): void {
+  if (typeof callback !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('callback', ['function'], callback);
+  }
+  try {
+    const result = fn();
```

Comment:
Shouldn't we execute this on the queue micro task, rather than doing sync work and returning the result in queue micro task?